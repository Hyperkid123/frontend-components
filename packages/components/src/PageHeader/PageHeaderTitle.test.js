import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import PageHeaderTitle from './PageHeaderTitle';

describe('PageHeader component', () => {
  it('should render', () => {
    const { container } = render(<PageHeaderTitle title="Something" />);
    expect(container).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    const title = 'Test Title';
    const actionsContent = 'Actions content';
    render(<PageHeaderTitle title={title} actionsContent={actionsContent} />);
    expect(screen.getByText(actionsContent)).toBeInTheDocument();
  });
});
