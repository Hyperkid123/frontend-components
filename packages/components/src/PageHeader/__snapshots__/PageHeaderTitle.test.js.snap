// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader component should render 1`] = `
<PageHeaderTitle
  title="Something"
>
  <Title
    className=""
    headingLevel="h1"
    size="2xl"
    widget-type="InsightsPageHeaderTitle"
  >
    <h1
      className="pf-v5-c-title pf-m-2xl"
      data-ouia-component-id="OUIA-Generated-Title-1"
      data-ouia-component-type="PF5/Title"
      data-ouia-safe={true}
      widget-type="InsightsPageHeaderTitle"
    >
      Something
    </h1>
  </Title>
</PageHeaderTitle>
`;
